### ISSUE TITLE
DataTypeService methods obsolete with no alternatives

### ISSUE THREAD

**Issue by matthewcare** [umbraco/Umbraco-CMS#17202]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

15.0.0-rc1

### Bug summary

There are a number of DataTypeService methods which are marked as obsolete, some stating removal in v15 and others v16.
There are, however, no alternatives for some marked obsolete, resulting in annoying build warnings.

### Specifics

My use case is that when creating new data types, I automatically categorise the data types into folders (containers) based on their alias.
e.g, 
```
private static string GetRelevantFolderName(IDataType dataType)
{
    return dataType.EditorAlias switch
    {
        "Umbraco.ContentPicker" => "Pickers",
        "Umbraco.MemberPicker" => "Pickers",
        "Umbraco.MemberGroupPicker" => "Pickers",
        "Umbraco.MultiNodeTreePicker" => "Pickers",
        "Umbraco.UserPicker" => "Pickers",

        "Umbraco.MultipleTextstring" => "Text",
        "Umbraco.TinyMCE" => "Text",
        "Umbraco.TextArea" => "Text",
        "Umbraco.TextBox" => "Text",

        "Umbraco.BlockGrid" => "Block Lists",
        "Umbraco.BlockList" => "Block Lists",

        "Umbraco.MediaPicker3" => "Images",

        _ => "Project"
    };
}
```

I am then using 
`dataTypeService.GetContainers(folderName, 1)`
to check if the folder exists, creating it if it doesn't, and using the ID to categorise the datatype.

There is no method which allows getting a container by its name or level, or any method of getting all or any container without already knowing its key in the proposed replacement, and using the `DataTypeRepository` directly feels incorrect

### Steps to reproduce

Look at the methods marked obsolete, and find an alternative in the service you are being told to replace with.

### Expected result / actual result

All methods being marked for removal need to have an alternative in the service described as the replacement.

**Comment by bergmania** [comment]
Thanks for letting us know. I have introduced some alternatives in https://github.com/umbraco/Umbraco-CMS/pull/17208

**Comment by nikolajlauridsen** [comment]
Fixed in #17208

Thank you for reporting 😄
